### PR TITLE
added capability of sharing without downloading

### PR DESCRIPTION
**Pull request by arjav2002**

### CODE REVIEW

**Comment by abhijeetviswa** [comment]
@arjav2002 You have 7 commits in this PR. Please squash the ones that are redundant and in the process, provide proper and meaningful commit messages. Also edit the PR message with the intention, and ideally link an issue to this PR. You may create one if an issue doesn't already exist.

File: app/src/main/java/helper/ModulesAdapter.java [hunk 1]
```diff
@@ -185,7 +189,21 @@ public void onClick(DialogInterface dialogInterface, int i) {
                                     mFileManager.downloadFile(module.getContents().get(0), module, courseName);
                                     break;
                                 case 1:
+                                    String toShare = "";
+                                    System.out.println(module.getContents());
```
Comment (SKrPl) [inline app/src/main/java/helper/ModulesAdapter.java]: Remove `System.out.println`. Please use [timber](https://github.com/JakeWharton/timber) for logging and debugging.

File: app/src/main/java/helper/ModulesAdapter.java [hunk 2]
```diff
@@ -185,7 +189,21 @@ public void onClick(DialogInterface dialogInterface, int i) {
                                     mFileManager.downloadFile(module.getContents().get(0), module, courseName);
                                     break;
                                 case 1:
+                                    String toShare = "";
```
Comment (SKrPl) [inline app/src/main/java/helper/ModulesAdapter.java]: Don't put the business logic here. Please create a private function and call it inside the switch-case.

File: app/src/main/res/layout/row_course_module_resource.xml [hunk 3]
```diff
@@ -1,132 +1,132 @@
...
+            android:paddingTop="6dp">
+
+            <LinearLayout
+                android:id="@+id/clickWrapperName"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_weight="1"
+                android:clickable="true"
+                android:focusable="true"
+                android:background="?android:selectableItemBackground"
```
Comment (SKrPl) [inline app/src/main/res/layout/row_course_module_resource.xml]: Except for this line, why have other lines been changed? I think you have changed the indentation type from tab to space or vice-versa. Please restore it as it was and only change the required lines.

File: gradle.properties [hunk 4]
```diff
@@ -13,6 +13,7 @@ android.enableJetifier=true
 android.useAndroidX=true
 org.gradle.jvmargs=-Xmx1536m
 
+
```
Comment (SKrPl) [inline gradle.properties]: Redundant change.
Ideally, this file should not be in VCS as it is a setting that would change from system to system. Can you please check in which PR (and see any specific reason to add this file) this file was added.

**Comment by SKrPl** [comment]
@arjav2002 I have left some comments, please go through them.

**Comment by arjav2002** [comment]
closing this pull request because the forked repo was deleted.

**Pull request closed without merging**